doc/source/reference/arrays.promotion.rst

```diff
@@ -0,0 +1,263 @@
+precision NumPy scalar will keep using its lower precision unless explicitly
+converted to a higher precision NumPy dtype or Python scalar (e.g. via ``int()``,
+``float()``, or ``scalar.item()``). This lower precision may be detrimental to
+some calculations or lead to incorrect results, especially in the case of integer
+overflows:
+
+  >>> np.int8(100) + 100  # the result exceeds the capacity of int8
+  np.int8(-56)
+  ... RuntimeWarning: overflow encountered in scalar add
+
+Note that NumPy warns when overflows occur for scalars, but not for arrays;
+e.g., ``np.array(100, dtype="uint8") + 100`` will *not* warn.
+
+Numerical promotion
+-------------------
+
+The following images shows the numerical promotion rules with the kinds
+on the vertical axis and the precision on the horizontal axis.
+
+.. figure:: figures/nep-0050-promotion-no-fonts.svg
+    :figclass: align-center
+
+Promotion is always found along the lines in the schema from left to right
+and down to up.  The final result dtype will smallest kind and precision
+(i.e. down and left) which is still equal or higher in both kind and precision
+while following the (green) lines.
+
+Note the following specific rules and observations:
+1. When a Python ``float`` or ``complex`` interacts with a NumPy integer
+   the result will be ``float64`` or ``complex128`` (yellow border).
+   NumPy booleans will also be cast to the default integer.[#default-int]
+   This is not relevant when additionally NumPy floating point values are
+   involved.
+2. The precision is drawn such that ``float16 < int16 < uint16`` because
+   large ``uint16`` do not fit ``int16`` and large ``int16`` will lose precision
+   when stored in a ``float16``.
+   This pattern however is broken since NumPy always considers ``float64``
+   and ``complex128`` to be acceptable promotion results for any integer
+   value.
+3. A special case the above leads to is that NumPy will promote many
+   combinations of signed and unsigned integers to ``float64`` because no
+   integer dtype can hold both inputs.
+   This can unfortunately be 
+
+The precision here comes from the bit size of the numerical value but
+an ``int32`` cannot always be stored in a ``float32`` without loss of
+precision, this leads to the 
+
+
+Notable or special promotion behaviors
+--------------------------------------
+
+In NumPy promotion refers to what specific functions do with the result and
+in some cases, this means that NumPy may deviate from what the `np.result_type`
+would give.
+
+Behavior of ``sum`` and ``prod``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+**``np.sum`` and ``np.prod``:** Will alway return the default integer type
+when summing over integer values (or booleans).  This is usually an ``int64``.
+The reason for this is that integer summations are otherwise very likely
+to overflow and give confusing results.
+This rule also applies to the underlying ``np.add.reduce``, ``np.multiply.reduce``
+and other reduction methods.
```

Comment:
Does it apply to all reductions over `int`, i.e., also `np.minimum.reduce` and `np.maximum.reduce`?

Reply:
no, those exact ones are the only special ones right now.

Reply:
Then it would seem best to delete "and other reduction methods".

Reply:
Deleted.  Maybe I thought of mean, or so... But those go to double anyway.